### ISSUE TITLE
Stealth skill

### ISSUE THREAD

**Issue by durson54** [Eniripsa96/SkillAPI#196]
Hey, so I want to ask something.
I have a skill name "Stealth" And it's like this:
Trigger: Cast
    Target: Self
       Condition: Combat - In Combat : False ; Seconds : 10
           Mechanic: Particle
           Mechanic: Sound
           Mechanic: Potion - Invisibility ; Seconds: 9999999
           Mechanic: Command - vanish
           Condition: Combat - In Combat: True ; Seconds : 10
               Mechanic: Command - vanish
               Mechanic: Purge - Potion: Invisibility ; Status: None
So basically it should do particle,sound,potion and command. - Working. But I want it that if he's in Stealth and he attacks someone it will activate Command again and Purge. So he'll no longer have vanish and potion and players will be able to attack him.

**Comment by AngDrew** [comment]
You know? I have better solution for you..
Just make physical attack trigger then remove their potion and play sound inside the physical attack trigger.. easier simpler

**Comment by AngDrew** [comment]
Using vanish is not effective one i think..

**Comment by durson54** [comment]
Thanks man, It's working :) But I want vanish so that they can't hit him if he's in Stealth and I added invisibility potion cuz I want to make skills only usabe while in Stealth so it'll use skill if he has invis potion. Hope you understand :) Thanks again

**Comment by durson54** [comment]
Oh, I know why vanish isn't effective :D Everytime I deal dmg it triggers vanish :D So I was thinking and i'll replace vanish with immunity :D

**Comment by AngDrew** [comment]
If you want cancel dmg add took physical damage and add condition when player affected by potion then add cancel mechanic as the children

**Comment by durson54** [comment]
Ok, thanks :) It's working as it should be now :)
